src/handlers/records-delete.ts

```diff
@@ -117,21 +119,28 @@ export class RecordsDeleteHandler implements MethodHandler {
       );
     }
   }
-}
-
-export async function constructIndexes(tenant: string, recordsDelete: RecordsDelete): Promise<Record<string, string>> {
-  const message = recordsDelete.message;
-  const descriptor = { ...message.descriptor };
-
-  // NOTE: the "trick" not may not be apparent on how a query is able to omit deleted records:
-  // we intentionally not add index for `isLatestBaseState` at all, this means that upon a successful delete,
-  // no messages with the record ID will match any query because queries by design filter by `isLatestBaseState = true`,
-  // `isLatestBaseState` for the initial delete would have been toggled to `false`
-  const indexes: Record<string, any> = {
-    // isLatestBaseState : "true", // intentionally showing that this index is omitted
-    author: recordsDelete.author,
-    ...descriptor
-  };
 
-  return indexes;
-}
+  /**
+  * Indexed properties needed for MessageStore indexing.
+  */
+  static constructIndexes(recordsDelete: RecordsDelete, recordsWrite: RecordsWriteMessage): Record<string, string> {
+    const message = recordsDelete.message;
+    const descriptor = { ...message.descriptor };
+    const { protocol, protocolPath, recipient, schema, parentId, dataFormat, dateCreated } = recordsWrite.descriptor;
```

Comment:
Can we add a comment here describing why these properties? I assume it's because these are the immutable properties?